File: media-service/internal/domain/entity/media.go
package entity

import "time"

type Media struct {
	ID               int64
	UserID           int64
	S3Key            string
	OriginalFilename string
	MimeType         string
	FileSize         int64
	IsMain           bool
	UploadedAt       time.Time
}

func NewMedia(userID int64, s3Key, originalFilename, mimeType string, fileSize int64, isMain bool) *Media {
	return &Media{
		UserID:           userID,
		S3Key:            s3Key,
		OriginalFilename: originalFilename,
		MimeType:         mimeType,
		FileSize:         fileSize,
		IsMain:           isMain,
		UploadedAt:       time.Now(),
	}
}

var AllowedMimeTypes = map[string]bool{
	"image/jpeg": true,
	"image/png":  true,
	"image/webp": true,
}

const MaxPhotosPerUser = 6

File: gateway-service/app/domain/media.py
from dataclasses import dataclass


@dataclass
class MediaItem:
    id: int
    user_id: int
    url: str
    original_filename: str
    mime_type: str
    file_size: int
    is_main: bool
    uploaded_at: str

    @classmethod
    def from_dict(cls, data: dict) -> "MediaItem":
        return cls(
            id=int(data.get("id", 0)),
            user_id=int(data.get("user_id", 0)),
            url=data.get("url", ""),
            original_filename=data.get("original_filename", ""),
            mime_type=data.get("mime_type", ""),
            file_size=int(data.get("file_size", 0)),
            is_main=bool(data.get("is_main", False)),
            uploaded_at=data.get("uploaded_at", ""),
        )
